boxoffice/views/admin_discount.py

```diff
@@ -0,0 +1,212 @@
+    discount_policy.items.extend(items)
+    db.session.commit()
+    if discount_policy_form.is_price_based.data:
+        discount_price = Price.query.filter_by(discount_policy=discount_policy).first()
+        discount_price.start_at = discount_policy_form.start_at.data
+        discount_price.end_at = discount_policy_form.end_at.data
+        discount_price.amount = discount_policy_form.amount.data
+        db.session.commit()
+    return api_success(result={'discount_policy': jsonify_discount_policy(discount_policy)}, doc="Discount policy updated.", status_code=200)
+
+
+@app.route('/api/1/admin/<org>/discount_policy/<discount_policy_id>/generate_coupon', methods=['OPTIONS', 'POST'])
+@lastuser.requires_login
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    (DiscountPolicy, {'id': 'discount_policy_id'}, 'discount_policy'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_create_coupon(organization, discount_policy):
+    coupon_form = DiscountCouponForm()
+    coupons = []
+    if not coupon_form.validate_on_submit():
+        return api_error(message=coupon_form.errors, status_code=400)
+    if coupon_form.count.data > 1:
+        # Create a signed discount coupon code
+        if not discount_policy.secret:
+            discount_policy.set_secret()
+        for x in range(coupon_form.count.data):
+            coupons.append(discount_policy.gen_signed_code())
+    else:
+        # Create a new discount coupon
+        if coupon_form.coupon_code.data:
+            # Custom discount code
+            coupon = DiscountCoupon(discount_policy=discount_policy,
+                usage_limit=coupon_form.usage_limit.data,
+                code=coupon_form.coupon_code.data)
+        else:
+            # Randomly generated discount code
+            coupon = DiscountCoupon(discount_policy=discount_policy,
+                usage_limit=coupon_form.usage_limit.data)
+        db.session.add(coupon)
+        db.session.commit()
+        coupons.append(coupon.code)
+    return api_success(result={'coupons': coupons}, doc="Discount coupon created.", status_code=200)
+
+
+@app.route('/api/1/admin/<org>/discount_policy/<discount_policy_id>/coupons')
+@lastuser.requires_login
+@load_models(
+    (DiscountPolicy, {'id': 'discount_policy_id'}, 'discount_policy'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_discount_coupons(discount_policy):
+    discount_coupons = DiscountCoupon.query.filter(DiscountCoupon.discount_policy == discount_policy).all()
+    coupons_list = []
+    for coupon in discount_coupons:
+        coupons_list.append({'code': coupon.code, 'usage_limit': coupon.usage_limit, 'available': coupon.usage_limit - coupon.used_count})
+    return api_success(result={'coupons': coupons_list}, doc="List of discount coupons", status_code=200)
+
+
+@app.route('/api/1/admin/discount_policy')
+@lastuser.requires_login
+@xhr_only
+@requestargs('discount_code_base')
+def admin_discount_policy(discount_code_base):
```

Comment:
What is this method for? Think it needs a better name. Also needs a permission check.